packages/aws-rfdk/lib/core/scripts/powershell/configureCloudWatchAgent.ps1

```diff
@@ -16,62 +16,90 @@ param (
-  } catch {
-    Write-Output "Failed to download CloudWatchAgent installer."
-    Exit 1
-  }
+  Write-Output "Failed to download CloudWatch agent installer."
+  Exit 1
+}
+
+$cwa_installer_sig = "$env:temp\amazon-cloudwatch-agent.msi.sig"
+try {
+  Read-S3Object -BucketName amazoncloudwatch-agent -Key windows/amd64/latest/amazon-cloudwatch-agent.msi.sig -File $cwa_installer_sig -Region us-east-1
+} catch {
+  Write-Output "Failed to download CloudWatch agent installer signature file."
+  Exit 1
 }
 
-# The below verification is tied to the version of the msi installer
-# If the installer version changes, the below verification could break and should be updated accordingly.
-# See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/verify-CloudWatch-Agent-Package-Signature.html
 if (-Not $s) {
-  $sig = Get-AuthenticodeSignature $file
-  $status = $sig | Select-Object -ExpandProperty 'Status'
+  # Download GPG
+  $gpg_installer = "$env:temp\gnupg-w32-2.2.27_20210111.exe"
+  wget https://gnupg.org/ftp/gcrypt/binary/gnupg-w32-2.2.27_20210111.exe -OutFile $gpg_installer
+
+  # Verify GPG
+  $gpg_sig = Get-AuthenticodeSignature $gpg_installer
+  $status = $gpg_sig | Select-Object -ExpandProperty 'Status'
   if ( $status -ne 'Valid' ) {
-    Write-Output "CloudWatchAgent installer does not have a valid signature."
+    Write-Output "GPG installer does not have a valid signature."
     Exit 1
   }
+  # Checksum information found at https://gnupg.org/download/integrity_check.html
+  $sha1 = Get-FileHash $gpg_installer -Algorithm SHA1
+  if ( $sha1 -inotmatch '9F2FF2CE36B6537F895AB3306527F105FF95DF8D') {
+    Write-Output "GPG failed checksum verification. Expected sha1 has to be '9F2FF2CE36B6537F895AB3306527F105FF95DF8D' but found:"
+    Write-Output $sha1
+    Exit 1
+  }
+
+  # Install GPG
+  Start-Process -Wait -FilePath $gpg_installer -ArgumentList "/S /v/qn" -PassThru
+
+  # Refresh the PATH so gpg is available
+  $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine")
 
-  # CA Certificate is:
-  #  - Subject: CN=DigiCert Timestamp Responder, O=DigiCert, C=US
-  #  - Valid: [Not After] 10/21/2024 7:00:00 PM
-  #  - Thumbprint: 614D271D9102E30169822487FDE5DE00A352B01D
-  $ca = $sig | Select-Object -ExpandProperty 'TimeStamperCertificate'
-  $ca_thumbprint = $ca | Select-Object -ExpandProperty 'Thumbprint'
-  if ( $ca_thumbprint -ne '614D271D9102E30169822487FDE5DE00A352B01D' ) {
-    Write-Output "CA Thumbprint failed verification. Has the CA certificate been changed? If so, then please submit a PR to fix this."
-    Write-Output $( $ca | Format-List )
+  # Download Amazon's public key and import it to GPG's keyring
+  $cloudwatch_pub_key = "$env:temp\amazon-cloudwatch-agent.gpg"
+  Read-S3Object -BucketName amazoncloudwatch-agent -Key assets/amazon-cloudwatch-agent.gpg -File $cloudwatch_pub_key -Region us-east-1
+  gpg --import $cloudwatch_pub_key
```

Comment:
Mind modifying this to use a temporary keyring instead of mutating the default keyring?
See the `--no-default-keyring` option usage in: https://github.com/aws/aws-rfdk/blob/b4f1d15ebe494f70b881e572a5d03e3fc46a3d95/packages/aws-rfdk/lib/core/scripts/bash/configureCloudWatchAgent.sh#L52

Reply:
Updated.